### PR TITLE
2.249.x

### PR DESCRIPTION
**Pull request by jglick**

### CODE REVIEW

File: failFast [hunk 1]
```diff
@@ -1 +1 @@
-true
+false
```
Comment (jglick) [inline failFast:1]: pending analysis of failures

**Comment by jglick** [comment]
`SnippetizerTest.oneOrMoreArgsStepDocs`, `BuildTriggerStepTest.buildStepDocs`, and `BuildTriggerStepTest.automaticParameterConversion` require https://github.com/jenkinsci/pipeline-build-step-plugin/pull/46 + https://github.com/jenkinsci/pipeline-input-step-plugin/pull/46 IIUC. See #297 for one of those.

**Comment by dwnusbaum** [comment]
`SnippetizerTest.oneOrMoreArgsStepDocs` requires https://github.com/jenkinsci/workflow-cps-plugin/pull/375.

`BuildTriggerStepTest.buildStepDocs` and `BuildTriggerStepTest.automaticParameterConversion` should be fixed if you update `pipeline-build-step` to 2.13.

`pipeline-input-step` 2.12 has some related fixed, but they only fix PCT issues in Blue Ocean and Pipeline Stage View, which are not in the BOM IIRC.

**Comment by jglick** [comment]
Hmm, now on plugins depending on `pipeline-build-step`—but not `pipeline-build-step` itself—I am getting

```
Failed to load: Pipeline: Build Step (2.13)
 - Update required: Pipeline: Supporting APIs (3.4) to be updated to 3.5 or higher
```

which does not make sense given https://github.com/jenkinsci/bom/blob/0575786c94652bb37382530a0e43961e8d909a31/bom-latest/pom.xml#L20. Some sort of bug in PCT? This release introduced usage of the BOM and I wonder if PCT is failing to override a version somewhere.

Also getting

```
com.gargoylesoftware.htmlunit.FailingHttpStatusCodeException: 403 Forbidden for http://localhost:37881/jenkins/
	at com.gargoylesoftware.htmlunit.WebClient.throwFailingHttpStatusCodeExceptionIfNecessary(WebClient.java:701)
	at com.gargoylesoftware.htmlunit.WebClient.loadDownloadedResponses(WebClient.java:2452)
	at com.gargoylesoftware.htmlunit.html.HtmlFormUtil.submit(HtmlFormUtil.java:82)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.login(JenkinsRule.java:2390)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.login(JenkinsRule.java:2257)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.login(JenkinsRule.java:2402)
	at org.jenkinsci.plugins.workflow.support.steps.input.InputStepTest.runAndContinue(InputStepTest.java:357)
	at org.jenkinsci.plugins.workflow.support.steps.input.InputStepTest.test_submitters(InputStepTest.java:238)
```

which needs to be evaluated; not obviously related to anything in https://github.com/jenkinsci/pipeline-input-step-plugin/compare/pipeline-input-step-2.11...pipeline-input-step-2.12.

**Comment by dwnusbaum** [comment]
@jglick I have also seen "Failed to load: ... Update required: Pipeline: Supporting APIs (3.4) to be updated to 3.5 or higher" for PCT runs against the latest release of `workflow-multibranch`, which also just started using the BOM.

**Comment by jglick** [comment]
Sigh, will have to dig into it. PCT’s POM editing is a pretty terrible bag of tricks. If I could find time to finish https://github.com/jenkinsci/maven-hpi-plugin/pull/66 that would be a much simpler and more reliable system.

**Comment by dwnusbaum** [comment]
@jglick `workflow-cps` 2.83 was just released with https://github.com/jenkinsci/workflow-cps-plugin/pull/375 to fix `SnippetizerTest.oneOrMoreArgsStepDocs`.

**Comment by jglick** [comment]
I think I tracked down the issue. When a plugin A (e.g., `credentials-binding`) has a BOM-managed dep on the `tests` JAR of another plugin B (e,g,. `workflow-support`), but does not explicitly mention B itself as being in the test classpath (because it already is, transitively from something else), all is fine normally—the BOM aligns the versions. The trouble comes when PCT edits `a/pom.xml`: it adds an explicit `version` to the `dependency` on B (with `classifier`) that it finds. Thus the `test`-scoped CP of A includes an older version of `b.hpi` alongside the newer version of `b-tests.jar`. This is not great, but usually harmless. The error comes when we run with `jth.jenkins-war.path` (as we do here) and there is _another_ plugin C (e.g., `pipeline-input-step`) in the megawar which depends on the new version of B: this will not load against the older version of B that it actually got, and in `InjectedTest`, `testPluginActive` fails because C did not load (even though this was not part of the test CP of B and may have been totally irrelevant to B’s tests).

**Comment by timja** [comment]
I think you’ll find a few of the ignored tests here are because of this issue

**Comment by jglick** [comment]
Timeout in `SCMRetrieverTest.selfTestLibraries` looks like an uncategorizable flake, maybe a problem with the build agent.

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**